src/app/tests/integration/chip_im_initiator.cpp

```diff
@@ -316,17 +306,9 @@ int main(int argc, char * argv[])
             goto exit;
         }
 
-        // Wait for response until the Message interval.
-        while (!MessageIntervalExpired())
-        {
-            DriveIO();
-        }
-
-        // Check if expected response was received.
-        if (gWaitingForCommandResp)
+        if (gCond.wait_for(lck, std::chrono::seconds(gMessageIntervalSeconds)) == std::cv_status::timeout)
```

Comment:
nit: In theory, you should use somethng like: `gCond.wait_for(lck, std::chrono::seconds(gMessageIntervalSeconds), [&] { return gReadRespCount > lastKnownCount; })` to handle spurious wakeups and the (quite unrealistic) case that `notify_one` is called before this thread enter the wait state.

Reply:
I agree, but for this simple test app, all requests are sent in sequence within the same thread, we should not hit the scenario you described above.